PersianCalendar@oxygenws.com/extension.js

```diff
@@ -277,6 +279,15 @@ const PersianCalendar = new Lang.Class({
         return true;
     },
 
+    _prayerTimeLoop: function(){
+        if (this._prayerTimeout) {
+            MainLoop.source_remove(this._prayerTimeout);
+            this._prayerTimeout = null;
+        }
+        this._prayerTimeout = MainLoop.timeout_add(60000 - (new Date().getSeconds() * 1000), Lang.bind(this,this._prayerTimeLoop));
```

Comment:
You should remove this in the "disable" function.
And I would suggest adding it to the "enable" function.

Another suggestion is to ignore adding timeout when the feature is disabled.